agent/sources/client/Makefile

```diff
@@ -0,0 +1,104 @@
+# cargo-style Makefile
+
+BUILD_DIR := ./target
+SRC_DIR := ./src
+TEST_DIR := ./tests
+
+EXEC_OBJ := main.o
+EXEC := rudder_client
+
+TEST_OBJ := run.o
+TEST_EXEC := test
+
+OUR_SRCS := $(shell find $(SRC_DIR) -name "*.[ch]" ! -path "*argtable3*")
+
+# main is handled separately
+SRCS := $(shell find $(SRC_DIR) ! -path *main.c -name *.c)
+OBJS := $(patsubst $(SRC_DIR)/%,$(BUILD_DIR)/%,$(SRCS:.c=.o))
+
+DESTDIR ?= $(BUILD_DIR)/dest
```

Comment:
all make variables are overidable by command line, is think at least those 2 those should not be overiden by evironment